### ISSUE TITLE
why the scores for Cross-Encoder Re-ranker are not normalize between 0 and 1.

### ISSUE THREAD

**Issue by abhibisht89** [UKPLab/sentence-transformers#1262]
Hi,

i am trying the demo give in the notebook for Retrieve & Re-Rank (retrieve_rerank_simple_wikipedia.ipynb)

I have one doubt why the scores for cross encoder are not normalize between 0 and 1.

![image](https://user-images.githubusercontent.com/13541524/141463338-0b021f5b-3831-4531-b458-f0d7dd970c62.png)

**Comment by nreimers** [comment]
It returns the logits scores unnormalized. If you want, you can pass them through a sigmoid function to get values 0-1. But this is not really needed, ranking results will not change

**Comment by abhibisht89** [comment]
@nreimers  thanks for the response. However i want to put the threshold on the cross encoder scores if it higher than the threshold then only show the match , else we can say there is not a good match. That's why i want to normalize scores. 

![image](https://user-images.githubusercontent.com/13541524/141478766-fa6bd34e-497f-450e-b3ad-c2a1674f31e2.png)

There is a pram : **apply_softmax** , which i thought i can use but this did not change the scores to probability. Not sure if this is the right way to use that param.

**Comment by nreimers** [comment]
You can also put a threshold on the raw logits. Instead of a threshold like 0.8, you use the inverse of sigmoid, so your threshold becomes e.g. 3.

Otherwise, you can call.
```
model = CrossEncoder(model_name, default_activation_function=nn.Sigmoid())
```

This will apply a sigmoid on top of the logits

**Comment by abhibisht89** [comment]
Thanks @nreimers for the response. i am working to implement this

**Comment by abhibisht89** [comment]
Thanks works well.

**Comment by horiacristescu** [comment]
Is using Softmax here wrong? Why sigmoid if the CrossEncoder has class ids as output?

**Comment by nreimers** [comment]
CrossEncoder outputs here a single value. Softmax would return always 1 if you only have one output value

**Comment by abhibisht89** [comment]
One thing that I want to mention is that , initially I  tries to put threshold on cross encode, however as @nreimers mention that it will not change the re ranking result.  Then I realize that the threshold  should be put on bi encoder result to get good candidate for cross encoder.  and if needed at the end we can put 1 one threshold on the cross encoder to filter result.
